internal/pkg/manifest/workload_ecs.go

```diff
@@ -91,15 +91,37 @@ type AlarmArgs struct {
 	MemoryUtilization *float64 `yaml:"memory_utilization"`
 }
 
-// DeploymentConfiguration represents the deployment strategies for a service.
+// WorkerAlarmArgs represents specs of CloudWatch alarms for Worker Service deployment rollbacks.
+type WorkerAlarmArgs struct {
+	AlarmArgs       `yaml:",inline"`
+	MessagesDelayed *int `yaml:"messages_delayed"`
+}
+
+// DeploymentControllerConfig represents deployment strategies for a service.
+type DeploymentControllerConfig struct {
+	Rolling *string `yaml:"rolling"`
+}
+
+// DeploymentConfiguration represents the deployment config for an ECS service.
 type DeploymentConfiguration struct {
-	Rolling        *string                    `yaml:"rolling"`
-	RollbackAlarms Union[[]string, AlarmArgs] // `yaml:"rollback_alarms"` 
+	DeploymentControllerConfig `yaml:",inline"`
+	RollbackAlarms             Union[[]string, AlarmArgs] // `yaml:"rollback_alarms"`
+	// The rollback_alarms manifest field is a no-op until the EDS-CFN ABR bug is fixed.
+}
+
+// WorkerDeploymentConfig represents the deployment strategies for a worker service.
+type WorkerDeploymentConfig struct {
+	DeploymentControllerConfig `yaml:",inline"`
+	WorkerRollbackAlarms       Union[[]string, WorkerAlarmArgs] // `yaml:"rollback_alarms"`
 	// The rollback_alarms manifest field is a no-op until the EDS-CFN ABR bug is fixed.
 }
 
 func (d *DeploymentConfiguration) isEmpty() bool {
-	return d == nil || (d.Rolling == nil && d.RollbackAlarms.IsZero())
+	return d == nil || (d.DeploymentControllerConfig.Rolling == nil && d.RollbackAlarms.IsZero())
```

Comment:
nit:
```suggestion
	return d == nil || (d.DeploymentControllerConfig.isEmpty() && d.RollbackAlarms.IsZero())
```
so that if we do have fields in addition to `rolling`, we won't have to remember to change the code here!